app/tests/test_routes.py

```diff
@@ -244,3 +246,121 @@ def test_route_generate_image_manifest(
+            ),
+            # standard user with only form data, no transferring_body_id in args
+            (
+                {"transferring_body_id": "form_value"},
+                {},
+                "main.search_transferring_body",
+                {"_id": "form_value"},
+            ),
+            # standard user with only args data, transferring_body_id present
+            (
+                {},
+                {"transferring_body_id": "args_value"},
+                "main.search_transferring_body",
+                {"_id": "args_value"},
+            ),
+        ],
+    )
+    def test_search_route_with_various_cases_standard_user(
+        app,
+        client: FlaskClient,
+        form_data,
+        args_data,
+        expected_redirect_route,
+        expected_params,
+        mock_standard_user,
+    ):
+        mock_standard_user(client)
+
+        query_string = "&".join(
+            [f"{key}={value}" for key, value in args_data.items()]
+        )
+        url = url_for("main.search") + "?" + query_string
+
+        response = client.get(url, data=form_data)
+        assert response.status_code == 302
+
+        if expected_redirect_route == "main.search_transferring_body":
+            redirected_url = url_for(
+                expected_redirect_route, _id=expected_params["_id"]
+            )
+        else:
+            redirected_url = url_for(expected_redirect_route, **expected_params)
+
+        assert redirected_url in response.headers["Location"]
+
+        for key, expected_value in expected_params.items():
+            assert f"{key}={expected_value}" in response.headers["Location"]
+
+    @pytest.mark.parametrize(
+        "form_data, args_data, expected_redirect_route, expected_params",
+        [
+            # all access user with args data (redirect to search_results_summary)
+            (
+                {},
+                {"some_param": "some_value"},
+                "main.search_results_summary",
+                {"some_param": "some_value"},
+            ),
+            # all access user with form data and args data (args takes precedence)
+            (
+                {"some_param": "form_value"},
+                {"some_param": "args_value"},
+                "main.search_results_summary",
+                {"some_param": "args_value"},
+            ),
+        ],
+    )
+    def test_search_route_with_various_cases_all_access_user(
+        app,
+        client: FlaskClient,
+        form_data,
+        args_data,
+        expected_redirect_route,
+        expected_params,
+        mock_all_access_user,
+    ):
+
+        mock_all_access_user(client)
+
+        query_string = "&".join(
+            [f"{key}={value}" for key, value in args_data.items()]
+        )
+        url = url_for("main.search") + "?" + query_string
+
+        response = client.get(url, data=form_data)
+        assert response.status_code == 302
+
+        if expected_redirect_route == "main.search_transferring_body":
```

Comment:
we still have conditionals in this test, can we split them out into individual test scenarios